airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py

```diff
@@ -56,29 +57,38 @@ def get_reports(self, config):
         try:
             session = requests.get(url, headers=auth.get_auth_header())
             body = session.json()
-            return [{"id": obj["id"], "type": obj["type"]} for obj in body]
+            return [obj["id"] for obj in body]
         except requests.exceptions.RequestException as e:
             return False, e
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         auth = self._get_authenticator(config)
 
+        default_start_date = datetime.now() - timedelta(days=2*365)
+        start_date = config.get("start_date", default_start_date.strftime('%Y-%m-%dT%H:%M:%S'))
+        day_page_size = config.get("day_page_size")
+
         result = [
             Feature(authenticator=auth),
             Guide(authenticator=auth),
             Page(authenticator=auth),
-            # Report(authenticator=auth),
+            Report(authenticator=auth),
             VisitorMetadata(authenticator=auth),
             AccountMetadata(authenticator=auth),
             Visitor(authenticator=auth),
             Account(authenticator=auth),
-            PageEvents(authenticator=auth),
-            FeatureEvents(authenticator=auth),
-            GuideEvents(authenticator=auth)
+            PageEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth),
+            FeatureEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth),
+            GuideEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth)
         ]
 
-        # all_reports = self.get_reports(config)
-        # for report in all_reports:
-        #     result.append(ReportResult(report=report, authenticator=auth))
+        all_reports = self.get_reports(config)
```

Comment:
no need to call `all_reports` here as you may not use it; you could move  into the else clause for an efficiency gain